app/controllers/spaces.rb

```diff
@@ -0,0 +1,90 @@
+class MakersBnb < Sinatra::Base
+  get '/spaces' do
+    @date_from = session[:date_from]
+    @date_to = session[:date_to]
+    @today = (Date.today).to_s
+    @tomorrow = (Date.today + 1).to_s
+
+    if @date_from && @date_to
+      @spaces = Space.all(:date_from.lte => @date_from,
+                          :date_to.gte => @date_to)
+    else
+      reset_date_filter
+      @spaces = Space.all
+    end
+    @user = User.get(session[:user])
+    erb :'spaces/spaces'
+  end
```

Comment:
Nice attempt, but again - far too much logic in the controller. Consider using a service object, or some model class functions, to decrease the complexity here.

> The perfect controller action is this: call a service, assign the result to a variable, and render a view using that variable. Three lines, tops
- _Some guy whose name I forgot_